### PR TITLE
Updated /users/:userId PATCH Endpoint

### PR DESCRIPTION
**Pull request by vikasosmium**
Date: 25-Dec-2024

Developer Name: [Vikas Singh](https://github.com/vikasosmium)

---

## Issue Ticket Number

- #2219 
## Description

This PR includes updating the PATCH route of Users /users/:userId and a deprecation warning for the old self route. This route will now allow superuser to approve the profilediff and authenticated user to update their profile data.
### Documentation Updated?

- [x] Yes
- [ ] No
- Doc Link https://github.com/Real-Dev-Squad/website-api-contracts/pull/211

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
**Unit Test**

Screenshot 2024-12-25 130515

**Integration Test**

image

## Test Coverage

**Unit Test Coverage**
Screenshot 2024-12-25 130401

**Integration Test Coverage**
image

### CODE REVIEW

File: test/integration/users.test.js [hunk 1]
```diff
@@ -1759,16 +1759,6 @@ describe("Users", function () {
...
-          first_name: "Ankur",
-          last_name: "Narkhede",
-          yoe: 0,
-          company: "",
-          designation: "AO",
-          github_id: "ankur1337",
-          linkedin_id: "ankurnarkhede",
-          twitter_id: "ankur909",
-          instagram_id: "",
-          website: "",
```
Comment (Achintya-Chatterjee) [inline test/integration/users.test.js:1771]: why did you deleted this ?

**Comment by vikasosmium** [comment]
We didn't need this while sending a patch request, it was unnecessarily
added in previous pr

**Review by Achintya-Chatterjee — APPROVED** [review]

File: controllers/users.js [hunk 2]
```diff
@@ -1096,6 +1103,26 @@ const updateUsernames = async (req, res) => {
...
+    const isSuperUser = roles[ROLES.SUPERUSER];
+    const profile = req.query.profile === "true";
+
+    if (isSelf && profile && req.query.dev === "true") {
+      return await updateSelf(req, res);
+    } else if (isSuperUser) {
+      return await updateUser(req, res);
+    }
+
+    return res.boom.unauthorized("You are not authorized for this action.");
```
Comment (pankajjs) [inline controllers/users.js:1119]: If isSelf and profile are true and dev is false which means it is a bad request,  then why are we sending 401 response?

Comment (vikasosmium) [reply controllers/users.js:1119]: done thanx!

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**